fastlane/CommonFastfile

```diff
@@ -0,0 +1,270 @@
+  content_after_opening_editor = File.read(changelog_path)
+  if content_before_opening_editor == content_after_opening_editor
+    unless UI.confirm("You may have opened the changelog in a visual editor. Enter 'y' when changes are saved or 'n' to cancel")
+      UI.user_error!("Cancelled")
+    end
+  end
+
+  changelog_path
+end
+
+desc "Creates new branch in the form `release/:version`. It receives `version` as parameter"
+private_lane :create_new_release_branch do |options|
+  release_version = options[:version]
+  UI.user_error!("missing version") unless release_version
+  sh("git checkout -b 'release/#{release_version}'")
+end
+
+private_lane :commit_updated_files_and_push do |options|
+  release_version = options[:version]
+  UI.user_error!("missing version") unless release_version
+  sh("git add -u")
+  sh("git commit -m 'Version bump for #{release_version}'")
+  push_to_git_remote
+end
+
+desc "Replace version number in project"
+lane :replace_version_number do |options|
+  new_version_number = options[:version]
+  UI.user_error!("missing version") unless new_version_number
+  files_to_update_string = ENV['FILES_TO_UPDATE_VERSION_STRING']
+  UI.user_error!("missing files to update env") unless files_to_update_string
+  files_to_update = files_to_update_string.split(",")
+  previous_version_number = current_version_number
+  
+  version_number_without_prerelease_modifiers = new_version_number.split("-")[0]
+  increment_version_number(version_number: version_number_without_prerelease_modifiers)  
+
+  for file_to_update in files_to_update
+    increment_build_number(previous_version_number, new_version_number, file_to_update.strip)
+  end
+end
+
+def attach_changelog_to_master(version_number)
+  current_changelog = File.open("../CHANGELOG.latest.md", 'r')
+  master_changelog = File.open("../CHANGELOG.md", 'r')
+
+  current_changelog_data = current_changelog.read
+  master_changelog_data = master_changelog.read
+
+  current_changelog.close  
+  master_changelog.close
+
+  File.open("../CHANGELOG.md", 'w') { |master_changelog_write_mode|
+    version_header = "## #{version_number}"
+    whole_file_data = "#{version_header}\n#{current_changelog_data}\n#{master_changelog_data}"
+    
+    master_changelog_write_mode.write(whole_file_data)
+  }
+end
+
+def increment_build_number(previous_version_number, new_version_number, file_path)
+  replace_in(previous_version_number, new_version_number, file_path)
+end
+
+def replace_in(previous_text, new_text, path, allow_empty=false)
+  if new_text.to_s.strip.empty? and not allow_empty
+    UI.user_error!("Missing `new_text` in call to `replace_in`, looking for replacement for #{previous_text} 😵.")
+  end
+  sed_regex = 's|' + previous_text.sub(".", "\\.") + '|' + new_text + '|'
+  backup_extension = '.bck'
+  sh("sed", '-i', backup_extension, sed_regex, path)
+end
+
+def current_version_number
+  File.read("../.version").strip
```

Comment:
Note how this checks for the existance of a `.version` file in the root folder. This is not the case in most of our repos. However, I think it's a good pattern to follow to have a common versioning system, since Android, iOS, Flutter,... each have their own versioning systems that would be tricky to abstract.

Reply:
💯💯💯